### PR TITLE
Feature/add services question areas question sets and questions question sets

### PR DESCRIPTION
**Pull request by GagarinPavel**

### CODE REVIEW

File: AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+
+        public QuestionAreasQuestionSetsServiceTests()
+        {
+            _service = new QuestionAreasQuestionSetsService(_repoMock.Object, _mapperMock.Object);
+        }
+
+        [Fact]
+        public async Task GetQuestionAreasQuestionSets_HasData_ReturnQuestionAreasQuestionSetsList()
+        {
+            List<QuestionAreasQuestionSetsEntity> questionAreasQuestionSetsEntities = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs:33]: Create additional static class and put it there

File: AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+            {
+                new()
+                {
+                    Id = 1,
+                    QuestionSetId = 1,
+                    QuestionAreaId = 1,
+                    QuestionArea = new()
+                    {
+                        Id = 1,
+                        Name = "sssss"
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs:43]: Maybe use real data?

**Review by nantonov — APPROVED** [review]

File: AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+            _repoMock.Setup(x => x.Get(default)).ReturnsAsync(questionAreasQuestionSetsEntities);
+            _mapperMock.Setup(x => x.Map<IEnumerable<QuestionAreasQuestionSets>>(It.IsAny<IEnumerable<QuestionAreasQuestionSetsEntity>>())).Returns(questionAreasQuestionSetsModels);
+            var result = await _service.Get(default);
+            Assert.Equal(new List<QuestionAreasQuestionSets>(), result);
+        }
+
+        [Fact]
+        public async Task GetQuestionAreasQuestionSetsById_ValidId_ReturnsQuestionAreasQuestionSetsById()
+        {
+            QuestionAreasQuestionSetsEntity questionAreasQuestionSetsEntity = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs:94]: Create additional static class and put it there

File: AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+                    Id = 1,
+                    Name = "sssss"
+                },
+                QuestionSet = new()
+                {
+                    Id = 1,
+                    Name = "ddddd"
+                }
+            };
+            QuestionAreasQuestionSets questionAreasQuestionSets = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs:110]: the same

**Review by CrazyKamenwik — APPROVED** [review]

File: AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+        {
+            _repoMock.Setup(x => x.Delete(int.MaxValue, default));
+
+            await _service.Delete(20, default).ShouldNotThrowAsync();
+        }
+
+        [Fact]
+        public async Task DeleteQuestionAreasQuestionSets_ValidQuestionAreasQuestionSets_ReturnsNull()
+        {
+            QuestionAreasQuestionSetsEntity questionAreasQuestionSetsEntity = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs:165]: the same

File: AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+                .Returns(questionAreasQuestionSetsEntity);
+            _repoMock.Setup(x => x.Delete(questionAreasQuestionSetsEntity, default));
+
+            await _service.Delete(20, default).ShouldNotThrowAsync();
+        }
+
+        [Fact]
+        public async Task PutQuestionAreasQuestionSets_ValidQuestionAreasQuestionSets_ReturnsQuestionAreasQuestionSets()
+        {
+            QuestionAreasQuestionSetsEntity questionAreasQuestionSetsEntity = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs:192]: the same

File: AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+            _repoMock.Setup(x => x.Update(questionAreasQuestionSetsEntity, default)).ReturnsAsync((QuestionAreasQuestionSetsEntity)null);
+            _mapperMock.Setup(x => x.Map<QuestionAreasQuestionSets>(It.IsAny<QuestionAreasQuestionSetsEntity>())).Returns((QuestionAreasQuestionSets)null);
+            var result = await _service.Put(null, default);
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public async Task AddQuestionAreasQuestionSets_ValidQuestionAreasQuestionSets_ReturnsQuestionAreasQuestionSets()
+        {
+            QuestionAreasQuestionSetsEntity questionAreasQuestionSetsEntity = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs:250]: the same

**Review by VadimZharikov — APPROVED** [review]

File: AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+            var actual = await _service.Add(questionAreasQuestionSets, default);
+
+            // Assert
+            Assert.Equal(questionAreasQuestionSetsWithId, actual);
+        }
+
+        [Fact]
+        public void GetQuestionAreasQuestionSetsByPredicate_ValidPredicate_ReturnsQuestionAreasQuestionSetsList()
+        {
+            List<QuestionAreasQuestionSetsEntity> questionAreasQuestionSetsEntities = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionAreasQuestionSetsServiceTests.cs:327]: the same

**Review by stas1408123 — APPROVED** [review]

File: AIS/AIS.BLL.Tests/Services/QuestionsQuestionSetsServiceTests.cs [hunk 9]
```diff
@@ -0,0 +1,405 @@
...
+
+        public QuestionsQuestionSetsServiceTests()
+        {
+            _service = new QuestionsQuestionSetsService(_repoMock.Object, _mapperMock.Object);
+        }
+
+        [Fact]
+        public async Task GetQuestionsQuestionSets_HasData_ReturnQuestionsQuestionSetsList()
+        {
+            List<QuestionsQuestionSetsEntity> questionsQuestionSetsEntities = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionsQuestionSetsServiceTests.cs:33]: the same

File: AIS/AIS.BLL.Tests/Services/QuestionsQuestionSetsServiceTests.cs [hunk 10]
```diff
@@ -0,0 +1,405 @@
...
+                .Returns(questionsQuestionSetsEntity);
+            _repoMock.Setup(x => x.Delete(questionsQuestionSetsEntity, default));
+
+            await _service.Delete(20, default).ShouldNotThrowAsync();
+        }
+
+        [Fact]
+        public async Task PutQuestionsQuestionSets_ValidQuestionsQuestionSets_ReturnsQuestionsQuestionSets()
+        {
+            QuestionsQuestionSetsEntity questionsQuestionSetsEntity = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionsQuestionSetsServiceTests.cs:192]: the same

File: AIS/AIS.BLL.Tests/Services/QuestionsQuestionSetsServiceTests.cs [hunk 11]
```diff
@@ -0,0 +1,405 @@
...
+            var actual = await _service.Add(questionsQuestionSets, default);
+
+            // Assert
+            Assert.Equal(questionsQuestionSetsWithId, actual);
+        }
+
+        [Fact]
+        public void GetQuestionsQuestionSetsByPredicate_ValidPredicate_ReturnsQuestionsQuestionSetsList()
+        {
+            List<QuestionsQuestionSetsEntity> questionsQuestionSetsEntities = new()
```
Comment (Blackolf931) [inline AIS/AIS.BLL.Tests/Services/QuestionsQuestionSetsServiceTests.cs:327]: the same

**Review by Blackolf931 — CHANGES_REQUESTED** [review]

**Pull request merged**